modules/custom/az_paragraphs/az_paragraphs_text_media/js/az_paragraphs_az_text_media_youtube.es6.js

```diff
@@ -0,0 +1,172 @@
+(($, Drupal) => {
+  Drupal.behaviors.az_youtube_video_bg = {
+    attach(context, settings) {
+      if (window.screen && window.screen.width > 768) {
+        // @see https://developers.google.com/youtube/iframe_api_reference
+        // defaults
+        const defaults = {
+          ratio: 16 / 9,
+          videoId: "",
+          mute: true,
+          repeat: true,
+          width: $(window).width(),
+          playButtonClass: "az-video-play",
+          pauseButtonClass: "az-video-pause",
+          muteButtonClass: "az-video-mute",
+          volumeUpClass: "az-video-volume-up",
+          volumeDownClass: "az-video-volume-down",
+          increaseVolumeBy: 10,
+          start: 0,
+          minimumSupportedWidth: 600
+        };
+        const { bgVideos } = settings.azFieldsMedia;
+        const bgVideoParagraphs = document.getElementsByClassName(
+          "az-js-video-background"
+        );
+        // load yt iframe js api
+        const tag = document.createElement("script");
+        const firstScriptTag = document.getElementsByTagName("script")[0];
+        tag.src = "https://www.youtube.com/iframe_api";
+        firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+        // methods
+        // set up iframe player, use global scope so YT api can talk
+        window.onYouTubeIframeAPIReady = () => {
+          $.each(bgVideoParagraphs, index => {
+            const thisContainer = bgVideoParagraphs[index];
+            const parentParagraph = thisContainer.parentNode;
+            const youtubeId = thisContainer.dataset.youtubeid;
+            bgVideos[youtubeId] = $.extend({}, defaults, thisContainer);
+            const options = bgVideos[youtubeId];
+            const videoPlayer = thisContainer.getElementsByClassName(
+              "az-video-player"
+            )[0];
+            const YouTubePlayer = window.YT;
+            thisContainer.player = new YouTubePlayer.Player(videoPlayer, {
+              width: options.width,
+              height: Math.ceil(options.width / options.ratio),
+              videoId: youtubeId,
+              playerVars: {
+                modestbranding: 1,
+                controls: 0,
+                showinfo: 0,
+                rel: 0,
+                wmode: "transparent"
+              },
+              events: {
+                onReady: window.onPlayerReady,
+                onStateChange: window.onPlayerStateChange
+              }
+            });
+            const PlayButton = bgVideoParagraphs[index].getElementsByClassName(
+              "az-video-play"
+            )[0];
+            PlayButton.addEventListener("click", event => {
+              event.preventDefault();
+              bgVideoParagraphs[index].player.playVideo();
+              parentParagraph.classList.remove("az-video-paused");
+              parentParagraph.classList.add("az-video-playing");
+            });
+            const PauseButton = bgVideoParagraphs[index].getElementsByClassName(
```

Comment:
More nitpicks I missed:
- I think it'd be better to use lowercase variable names for `PauseButton` and `PlayButton` for consistency.  The `YouTubePlayer` variable is actually an object so it might make sense to keep that one capitalized...

Reply:
yeah YouTubePlayer comes from the API, good call